### PR TITLE
Add support for projects_v2 and projects_v2_item webhook events

### PR DESCRIPTION
**Pull request by nikpivkin**
Closes https://github.com/google/go-github/issues/2867

Added support for [projects_v2](https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#projects_v2) and [projects_v2_item](https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#projects_v2_item) webhook events.

### CODE REVIEW

**Comment by gmlewis** [comment]
This PR has significant conflicts with #2865 due to said PR dramatically changing the internals for webhook events (which will hopefully also reduce the maintenance burden and increase the simplicity of future PRs like this one).

So I apologize for the timing.

We can either move forward with this PR and have more work required in #2865 or we can move forward with #2865 then require changes here.

I'm honestly fine either way.

Do you have a preference @nikpivkin ?
Or do you have a preference, @evankanderson?

**Comment by evankanderson** [comment]
I'm happy either way; I'll be updating my PR tomorrow probably

File: github/event_types.go
Comment (gmlewis) [inline github/event_types.go:946]: nit: Please add punctuation to godoc-style comments.

```suggestion
// The Webhook event name is "projects_v2".
```

File: github/event_types.go
Comment (gmlewis) [inline github/event_types.go:977]:
```suggestion
// ProjectV2ItemEvent is triggered when there is activity relating to an item on an organization-level project.
```

File: github/event_types.go
Comment (gmlewis) [inline github/event_types.go:978]:
```suggestion
// The Webhook event name is "projects_v2_item".
```

File: github/event_types_test.go
Comment (gmlewis) [inline github/event_types_test.go:14303]: Please remove blank line.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @nikpivkin !
Just a few minor tweaks, please before proceeding.

**Review by gmlewis — APPROVED** [review]
Thank you, @nikpivkin !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.
(Maybe @evankanderson has time so that this can be merged before work is continued on #2865 ?)

**Review by evankanderson — APPROVED** [review]

**Comment by evankanderson** [comment]
Not sure if my review/approval helps, but LGTM.  Double checked that the types seem to match the documentation.

**Comment by gmlewis** [comment]
> Not sure if my review/approval helps, but LGTM. Double checked that the types seem to match the documentation.

Yes it does, as you are a valued contributor to this repo.  Thank you, @evankanderson !
Merging.

**Pull request merged**
